privacyidea/lib/auditmodules/sqlaudit.py

```diff
@@ -116,7 +116,10 @@ def _truncate_data(self):
         """
         for column, l in column_length.iteritems():
             if column in self.audit_data:
-                self.audit_data[column] = self.audit_data[column][:l]
+                data = self.audit_data[column]
+                if isinstance(data, basestring):
```

Comment:
One remaining point: If the audit data contains a boolean value, we don't touch it. This may cause problems for Oracle DBs like in #1000, because a ``bool`` value is written to a ``db.String`` column.